server/routes/topics.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const router = express.Router();
+const User = require('../database/models/user');
+
+router.post('/', (req, res, next) => {
+    User.findOne({ topics: req.body.entryTerm }, (err, topic) => {
```

Comment:
Order of operations to allow a user to add a new topic to their list:
1. Identify the user by the `username`. Every server request (since Auth v2 changes) now includes an authorization token that is validated & decoded with the `cognito:username`. See the User route for an example of how this works.
1. Check if the new topic already exists for that user. If so, return an error code with `res.sendStatus(409)`. The client could probably gracefully handle this error by doing the same exact thing as if the new topic was added succesfully.
1. If the topic does not exist, add it to the database and return the new list of topics. I'm not sure exactly how Mongo works in this case, but I'm guessing `dbTopics` actually represents a User object here.

Reply:
My comments / questions (in a somewhat random order):
1. Do I need to identify the user since he/she can't get to this screen without being logged in? The add / delete topics operations take place with the user's Mongo document. If they're not logged in, those operations have no place to go. Not trying to avoid your Step 1, just trying to understand why it's necessary. 
2. Your point about gracefully handling the error makes sense (and resolves my question above): the user doesn't need to know that they forgot that they already put that topic into their database. We have to make sure there are no duplicates, but they don't need to go through an extra step - whether or not the topic exists, we return a `dbTopics` that contains that item. 
3. Yes, `dbTopics` is the object that gets plugged into the profile

Reply:
1. Let’s discuss over Zoom. Yes, we must identify the user somehow to know which user’s topics to update. Grabbing the username from the authorization token is a secure way to do this.

3. I ask about the variable name “dbTopics” because if it actually represents a user object, then I’d rename it to “user” to avoid confusion.